test/system_SUITE.erl

```diff
@@ -158,6 +161,145 @@ validation_failure_for_AMQP_client1(Config) ->
+
+    ok = whitelist(Config, "alice", CertTrusted,  KeyTrusted),
+    ok = change_configuration(rabbitmq_trust_store, [{directory, whitelist_dir(Config)}]),
+
+    ok = rabbit_networking:start_ssl_listener(Port, [{cacerts, [Root]},
+                                                     {cert, Cert},
+                                                     {key, Key} | cfg()], 1),
+
+    %% When: a client connects and present `CertInter` as well as the `CertTrusted`
+    %% Then: the connection is successful.
+    {ok, Con} = amqp_connection:start(#amqp_params_network{host = Host,
+                                                           port = Port,
+                                                           ssl_options = [{cacerts, [CertInter]},
+                                                                          {cert, CertTrusted},
+                                                                          {key, KeyTrusted}]}),
+
+    %% When: a client connects and present `RootTrusted` and `CertInter` as well as the `CertTrusted`
+    %% Then: the connection is successful.
+    {ok, Con2} = amqp_connection:start(#amqp_params_network{host = Host,
+                                                            port = Port,
+                                                            ssl_options = [{cacerts, [RootCA, CertInter]},
+                                                                           {cert, CertTrusted},
+                                                                           {key, KeyTrusted}]}),
+
+    %% When: a client connects and present `CertInter` and `RootCA` as well as the `CertTrusted`
+    %% Then: the connection is successful.
+    {ok, Con3} = amqp_connection:start(#amqp_params_network{host = Host,
+                                                            port = Port,
+                                                            ssl_options = [{cacerts, [CertInter, RootCA]},
+                                                                           {cert, CertTrusted},
+                                                                           {key, KeyTrusted}]}),
+
+    % %% When: a client connects and present `CertInter` and `RootCA` but NOT `CertTrusted`
+    % %% Then: the connection is not succcessful
+    {error, ?SERVER_REJECT_CLIENT} =
+        amqp_connection:start(#amqp_params_network{host = Host,
+                                                   port = Port,
+                                                   ssl_options = [{cacerts, [RootCA]},
+                                                                  {cert, CertInter},
+                                                                  {key, KeyInter}]}),
+
+    %% When: a client connects and present `CertUntrusted` and `RootCA` and `CertInter`
+    %% Then: the connection is not succcessful
+    %% TODO: for some reason this returns `bad certifice` rather than `unknown ca`
+    {error, {tls_alert, "bad certificate"}} =
```

Comment:
This makes certain sense to me: the peer certificate is "bad" (untrusted) so this can be reported before "unknown" (untrusted) CA.